modules/ui/photoviewer.js

```diff
@@ -75,30 +75,34 @@ export function uiPhotoviewer(context) {
 
 
         function setPhotoFromViewerButton() {
-            if (services.mapillary.isViewerOpen()) {
-                if (context.mode().id !== 'select' || !(layerStatus('mapillary') && getServiceId() === 'mapillary')) {
+            const service = getServiceId();
+            const isActiveService = ['mapillary', 'panoramax'].includes(service) && services[service].isViewerOpen();
+            const mapillaryOrPanoramax = (layerStatus(service) && ['mapillary', 'panoramax'].includes(service));
```

Comment:
- Lets use a more generic name for the const
- Lets reuse the same array defined once above

Can you double check the conditions, I think that isActiveService already accounts for the second condition in this check, right?